src/Transform/ONNX/ShapeInferencePass.cpp

```diff
@@ -34,7 +35,7 @@ class ShapeInferencePass
     // Iterate on the operations that need shape inference i.e the operations
     // that return a dynamic shape.
     f.walk([&](mlir::Operation *op) {
-      if (returnsDynamicShape(op)) {
+      if (isUsedByReturnOp(op) || returnsDynamicShape(op)) {
```

Comment:
May want to record why this is needed, it's not entirely obvious.

Reply:
@tjingrant Thanks for your review! I added some comments here